pkg/controller/networkpolicy/clusternetworkpolicy.go

```diff
@@ -117,88 +121,210 @@ func (n *NetworkPolicyController) deleteCNP(old interface{}) {
 	n.deleteDereferencedAddressGroups(oldInternalNP)
 }
 
+// filterPerNamespaceRuleACNPsByNSLabels gets all ClusterNetworkPolicy names that will need to be
+// re-processed if a Namespace adds or removes the input labels.
+func (n *NetworkPolicyController) filterPerNamespaceRuleACNPsByNSLabels(nsLabels labels.Set) sets.String {
```

Comment:
If the input labels are not entire label set of the namespace, can it work correctly?
For example, if a selector is "app=foo, tier=frontend", and a namespace is updated from "app=foo, tier=frontend" to "app=foo", the input label is "tier=fronend", which can't match the selector, right?

Reply:
Thanks for spotting this! Fixed and will add a test for this scenario

Reply:
Yes. If you look at the `updateNamespace` func now, it uses the entire label set of the Namespace before the update and after the update, so the scenario you pointed out above is no longer an issue.